### PR TITLE
ui replay: add mici UI exploration

### PR DESCRIPTION
**Pull request by TheSecurityDev**
Adds support for dragging/swiping in replays and completely overhauls the simple mici script with an almost 90-second one that tests much of the settings as well as onroad and offroad alerts

Still many edge cases that could be explored further, such as more of the training guide, the DM camera preview, and also different states of the settings such as while onroad, or paired vs unpaired, etc.

Should be pretty easy to update in the future if things change.

### CODE REVIEW

File: selfdrive/ui/tests/diff/replay_script.py [hunk 1]
```diff
@@ -61,26 +68,47 @@ def wait(self, frames: int) -> None:
...
+    # Interpolate positions over duration_frames
+    for i in range(1, duration_frames):
+      t = i / duration_frames
+      x, y = int(start_x + delta_x * t), int(start_y + delta_y * t)
+      mouse_move = MouseEvent(pos=MousePos(x, y), slot=0, left_pressed=False, left_released=False, left_down=True, t=self.get_frame_time())
+      self.add(ScriptEvent(mouse_events=[mouse_move]), after=1)
+
+    # Mouse up at end
+    mouse_up = MouseEvent(pos=MousePos(end_x, end_y), slot=0, left_pressed=False, left_released=True, left_down=False, t=self.get_frame_time())
+    self.add(ScriptEvent(mouse_events=[mouse_up]), after=wait_after)
```
[the remark this answers is not in the document]
Comment (TheSecurityDev) [reply selfdrive/ui/tests/diff/replay_script.py:110]: Hmmm that would likely mess up whatever calculations have already been set for the script

**Comment by sshane** [comment]
nice! did we make the diff replay deterministic yet? the nav bar used to sometimes fail

**Comment by TheSecurityDev** [comment]
> nice! did we make the diff replay deterministic yet? the nav bar used to sometimes fail

Yep, that should have been fixed in #37110. I haven't seen any issues since then. The only remaining source of indeterminism might be for animations that use `time.monotonic`, but we can mock that as well later if necessary.
